profitbricks/src/main/java/org/jclouds/profitbricks/features/SnapshotApi.java

```diff
@@ -0,0 +1,86 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.profitbricks.features;
+
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.Fallbacks;
+import org.jclouds.http.filters.BasicAuthentication;
+import org.jclouds.profitbricks.binder.snapshot.CreateSnapshotRequestBinder;
+import org.jclouds.profitbricks.binder.snapshot.RollbackSnapshotRequestBinder;
+import org.jclouds.profitbricks.binder.snapshot.UpdateSnapshotRequestBinder;
+import org.jclouds.profitbricks.domain.Snapshot;
+import org.jclouds.profitbricks.http.filters.ProfitBricksSoapMessageEnvelope;
+import org.jclouds.profitbricks.http.parser.RequestIdOnlyResponseHandler;
+import org.jclouds.profitbricks.http.parser.snapshot.SnapshotResponseHandler;
+import org.jclouds.profitbricks.http.parser.snapshot.SnapshotListResponseHandler;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.Payload;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.XMLResponseParser;
+
+@RequestFilters({BasicAuthentication.class, ProfitBricksSoapMessageEnvelope.class})
+@Consumes(MediaType.TEXT_XML)
+@Produces(MediaType.TEXT_XML)
+public interface SnapshotApi {
+
+   @POST
+   @Named("snapshot:getall")
+   @Payload("<ws:getAllSnapshots/>")
+   @XMLResponseParser(SnapshotListResponseHandler.class)
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<Snapshot> getAllSnapshots();
+
+   @POST
+   @Named("snapshot:get")
+   @Payload("<ws:getSnapshot><snapshotId>{snapshotId}</snapshotId></ws:getSnapshot>")
+   @XMLResponseParser(SnapshotResponseHandler.class)
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   Snapshot getSnapshot(@PayloadParam("snapshotId") String identifier);
+
+   @POST
+   @Named("snapshot:create")
+   @MapBinder(CreateSnapshotRequestBinder.class)
+   @XMLResponseParser(SnapshotResponseHandler.class)
+   Snapshot createSnapshot(@PayloadParam("snapshot") Snapshot.Request.CreatePayload payload);
+
+   @POST
+   @Named("snapshot:update")
+   @MapBinder(UpdateSnapshotRequestBinder.class)
+   @XMLResponseParser(RequestIdOnlyResponseHandler.class)
+   String updateSnapshot(@PayloadParam("snapshot") Snapshot.Request.UpdatePayload payload);
+
+   @POST
+   @Named("snapshot:delete")
+   @Payload("<ws:deleteSnapshot><snapshotId>{id}</snapshotId></ws:deleteSnapshot>")
+   @Fallback(Fallbacks.FalseOnNotFoundOr404.class)
+   boolean deleteSnapshot(@PayloadParam("id") String id);
+
+   @POST
+   @Named("snapshot:rollback")
+   @MapBinder(RollbackSnapshotRequestBinder.class)
+   boolean rollbackSnapshot(@PayloadParam("snapshot") Snapshot.Request.RollbackPayload payload);
```

Comment:
Actually I'm looking right now into my .net library and it never returns 404. Since it is SOAP service it is returning 500 and fault xml.
The thing is I was following how it is done in other features like servers https://github.com/StackPointCloud/jclouds-labs/blob/master/profitbricks/src/main/java/org/jclouds/profitbricks/features/ServerApi.java